src/hotspot/share/gc/shenandoah/shenandoahBarrierSet.inline.hpp

```diff
@@ -121,10 +121,12 @@ inline oop ShenandoahBarrierSet::load_reference_barrier(DecoratorSet decorators,
     return nullptr;
   }
 
-  // Prevent resurrection of unreachable objects that are visited during
-  // concurrent class-unloading.
+  // Allow runtime to see unreachable objects that are visited during concurrent class-unloading.
+  // Note that this may also interfere with the DeadCounterClosure when visiting weak oop storage,
+  // but it does not seem to be a problem in practice because the dead count callbacks do not care
+  // about the precise number of dead objects (only that there are dead objects).
```

Comment:
The last 3 lines feel too specific, really :) I think that paragraph should be in the related bug report.

Reply:
Okay.